src/components/DeviceStatusGauge.js

```diff
@@ -0,0 +1,241 @@
+import { h } from 'preact'
+import _ from 'lodash'
+import Pie from 'paths-js/pie'
+import styled, { withTheme } from 'styled-components'
+import { compose } from 'recompose'
+import ResinDeviceStatus from 'resin-device-status'
+
+import hoc from '../hoc'
+import theme from '../theme'
+
+const { colors, font } = theme
+
+const DeviceStatusGaugeWrapper = styled.div`
+  display: block;
+
+  .device-status-gauge__graph {
+    display: block;
+    width: 100%;
+    max-width: 180px;
+    margin: 0 auto 10px;
+  }
+
+  .device-status-gauge__graph__total-devices-label {
+    font-family: ${font};
+    font-size: 5px;
+    fill: #b8b8b8;
+    text-anchor: middle;
+  }
+
+  .device-status-gauge__graph__total-devices-number {
+    font-family: ${font};
+    font-size: 26px;
+    fill: #5e5e5e;
+    letter-spacing: -1px;
+    text-anchor: middle;
+  }
+
+  .device-status-gauge__statuses {
+    .clearfix;
+    list-style: none;
+    padding: 5px;
+    margin: 0 auto;
+    background: white;
+    border-radius: 5px;
+    max-width: 240px;
+    border: 1px solid #ececec;
+    overflow: hidden;
+  }
+
+  .device-status-gauge__status {
+    display: flex;
+    padding: 3px 0 3px 2px;
+    line-height: 9px;
+    width: 50%;
+    float: left;
+  }
+
+  .device-status-gauge__status--post-provisioning {
+    width: 100%;
+  }
+
+  .device-status-gauge__status__color {
+    display: inline-block;
+    flex: 0 0 auto;
+    width: 10px;
+    height: 10px;
+    margin-right: 4px;
+  }
+
+  .device-status-gauge__status__name {
+    font-size: 11px;
+  }
+  .block-status-idle {
+    background-color: ${colors.statusIdle.main};
+  }
+  .block-status-configuring {
+    background-color: ${colors.statusConfiguring.main};
+  }
+  .block-status-updating {
+    background-color: ${colors.statusUpdating.main};
+  }
+  .block-status-post-provisioning {
+    background-color: ${colors.statusPostProvisioning.main};
+  }
+  .block-status-offline {
+    background-color: ${colors.statusOffline.main};
+  }
+`
+
+const getDefaultPlaceholderState = () => ({
+  count: 0,
+  key: ResinDeviceStatus.status.OFFLINE,
+  name: _.get(
+    ResinDeviceStatus.statuses.find(
+      s => s.key === ResinDeviceStatus.status.OFFLINE
+    ),
+    'name',
+    'Offline'
+  ),
+  value: 1
+})
+
+const pathFillColor = status => {
+  if (status === 'idle') {
```

Comment:
this could be `return colors[`status${camelCase(status)}`].main`

Reply:
But in IMO this should just be called warning, info, ... then etcher and other libs could use the gauge with out it being device specific.